src/main/java/jenkins/branch/MultiBranchProjectDisplayNamingStrategy.java

```diff
@@ -60,7 +60,14 @@ public String generateName(@NonNull final String rawName, final String displayNa
                 return rawName;
             }
 
-            return format("%s - %s", rawName, displayName);
+            // The raw name provided here in the context of pull requests is the pull request ID
+            // We tidy up the ID so that they display consistently between SCMs
+            String cleanedUpBranchName = rawName;
+            if (cleanedUpBranchName.startsWith("MR-") || cleanedUpBranchName.startsWith("PR-")) {
```

Comment:
this would appear to be putting SCM specifics into an SCM agnostic piece of code?
I believe other systems exist and Gerrit uses `C-xxxx`

Reply:
Is this a blocker, and if so would this mean an API likely needs to be created where the SCMs set their prefix?

Reply:
So initially I was not thinking it was a blocker, but something that should be addressed.

But now I am thinking is this just confusing?  we use `#nnn` for build numbers in Jenkins and do I really want or need consistency across SCMs?   Indeed having PR-123 in one job and MR-245 in another is a reminder that one project is in my GitHub.com account  and the other is in GitLab on premise.

Reply:
I think in context with more than one pull request it will be really clear and not going to get confusing at all, e.g.:

<img width="251" alt="image" src="https://github.com/user-attachments/assets/193b8a8b-3ad4-45f5-8e18-f387e47ea192">

I don't think most users will care where the SCM is when looking at a job.